modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManager.java

```diff
@@ -953,16 +971,23 @@ public SCIMResponse updateWithPATCH(String existingGroupId, String patchRequest,
             SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
             Map<String, Boolean> requestAttributes = ResourceManagerUtil.getAllAttributeURIs(schema);
 
-            Group oldGroup = userManager.getGroup(existingGroupId, requestAttributes);
-            if (oldGroup == null) {
-                throw new NotFoundException("No group with the id : " + existingGroupId + " exists in the user store.");
-            }
+            List<PatchOperation> opList = getDecoder().decodeRequest(patchRequest);
 
-            // Make a copy of original group. This will be used to restore to the original condition if failure occurs.
-            Group originalGroup = (Group) CopyUtil.deepCopy(oldGroup);
-            Group patchedGroup = doPatchGroup(oldGroup, schema, patchRequest);
+            if (!isDeleteAllUsersOperationFound(opList)) {
+                updateWithPatchForAddRemoveOperations(existingGroupId, opList, userManager);
+            } else {
+                Group oldGroup = userManager.getGroup(existingGroupId, requestAttributes);
+                if (oldGroup == null) {
+                    throw new NotFoundException("No group with the id : " + existingGroupId
```

Comment:
Shall we use String.format util function to concatenate strings.

Reply:
What is the benefit in using String.format here since only one param is passed here

Reply:
@mpmadhavig Refer https://github.com/wso2/carbon-kernel/pull/2748#discussion_r516516308